Missing Endpoint does not result in errored event or retry

[ERROR] 12:20:42.305 [nats:16] nl.nuts.consent.bridge.pipelines.NatsToCordaPipeline - Not all given organizations have a consent endpoint in the registry
java.lang.IllegalArgumentException: Not all given organizations have a consent endpoint in the registry
	at nl.nuts.consent.bridge.corda.CordaService.createConsentBranch(CordaService.kt:355) ~[classes!/:?]
	at nl.nuts.consent.bridge.pipelines.NatsToCordaPipeline.processEventConsentRequestConstructed(NatsToCordaPipeline.kt:223) ~[classes!/:?]
	at nl.nuts.consent.bridge.pipelines.NatsToCordaPipeline.processEvent(NatsToCordaPipeline.kt:188) ~[classes!/:?]
	at nl.nuts.consent.bridge.pipelines.NatsToCordaPipeline$startListeners$1.onMessage(NatsToCordaPipeline.kt:126) ~[classes!/:?]
	at io.nats.streaming.StreamingConnectionImpl.processMsg(StreamingConnectionImpl.java:932) ~[java-nats-streaming-2.2.3.jar!/:2.2.3]
	at io.nats.streaming.StreamingConnectionImpl.lambda$getDispatcherByName$4(StreamingConnectionImpl.java:527) ~[java-nats-streaming-2.2.3.jar!/:2.2.3]
	at io.nats.client.impl.NatsDispatcher.run(NatsDispatcher.java:98) ~[jnats-2.6.7.jar!/:2.6.7]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]

--- a/impl/src/main/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipeline.kt
+++ b/impl/src/main/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipeline.kt
@@ -132,14 +132,14 @@ class NatsToCordaPipeline {
                     retry(event)
                 } catch (e: Exception) { // broken
                     logger.error(e.message, e)
-                    publish(NATS_CONSENT_ERROR_SUBJECT, it.data)
+                    publishError(event, e.message)
                 }
             } catch (e: JsonParseException) { // broken
                 logger.error(e.message, e)
-                publish(NATS_CONSENT_ERROR_SUBJECT, it.data)
+                publishError(it.data, e.message)
             } catch (e: JsonMappingException) { // broken
                 logger.error(e.message, e)
-                publish(NATS_CONSENT_ERROR_SUBJECT, it.data)
+                publishError(it.data, e.message)
             } catch (e: Exception) { // broken
                 logger.error(e.message, e)
             }
@@ -176,6 +176,25 @@ class NatsToCordaPipeline {
         val newEvent = e.copy(retryCount = e.retryCount + 1)
         val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
         retry(newEvent.retryCount, bytes)
+    }
+
+    private fun publishError(e: Event, message: String?) {
+        val newEvent = e.copy(name = EventName.EventErrored, error = message)
+        val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
+        publish(NATS_CONSENT_ERROR_SUBJECT, bytes)
+    }
+
+    private fun publishError(erroredBytes: ByteArray, message: String?) {
+        val newEvent = Event(
+            name = EventName.EventErrored,
+            error = "$message, event bytes in payload",
+            payload = String(erroredBytes),
+            UUID = UUID.randomUUID().toString(),
+            externalId = "",
+            retryCount = 0
+        )
+        val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
+        publish(NATS_CONSENT_ERROR_SUBJECT, bytes)
     }
 
     /**

--- a/impl/src/main/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipeline.kt
+++ b/impl/src/main/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipeline.kt
@@ -173,28 +173,27 @@ class NatsToCordaPipeline {
     }
 
     private fun retry(e: Event) {
-        val newEvent = e.copy(retryCount = e.retryCount + 1)
-        val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
+        val newEvent= e.copy(retryCount = e.retryCount + 1)
+        val bytes= Serialization.objectMapper().writeValueAsBytes(newEvent)
         retry(newEvent.retryCount, bytes)
     }
 
     private fun publishError(e: Event, message: String?) {
-        val newEvent = e.copy(name = EventName.EventErrored, error = message)
-        val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
+        val newEvent= e.copy(name = EventName.EventErrored, error = message)
+        val bytes= Serialization.objectMapper().writeValueAsBytes(newEvent)
         publish(NATS_CONSENT_ERROR_SUBJECT, bytes)
     }
 
     private fun publishError(erroredBytes: ByteArray, message: String?) {
-        val newEvent = Event(
+        val e = Event(
             name = EventName.EventErrored,
-            error = "$message, event bytes in payload",
+            error = "",
             payload = String(erroredBytes),
             UUID = UUID.randomUUID().toString(),
             externalId = "",
             retryCount = 0
         )
-        val bytes  = Serialization.objectMapper().writeValueAsBytes(newEvent)
-        publish(NATS_CONSENT_ERROR_SUBJECT, bytes)
+        publishError(e, "$message, event bytes in payload")
     }
 
     /**

--- a/impl/src/test/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipelineIntegrationTest.kt
+++ b/impl/src/test/kotlin/nl/nuts/consent/bridge/pipelines/NatsToCordaPipelineIntegrationTest.kt
@@ -55,6 +55,7 @@ import java.time.OffsetDateTime
 import java.util.*
 import kotlin.collections.ArrayList
 import kotlin.test.assertEquals
+import kotlin.test.assertTrue
 
 
 class NatsToCordaPipelineIntegrationTest {
@@ -169,13 +170,17 @@ class NatsToCordaPipelineIntegrationTest {
         Thread.sleep(2000)
         // assert
         assertEquals(1, erroredEvents.size)
+
+        val e: Event = Serialization.objectMapper().readValue(erroredEvents.first().byteInputStream(), Event::class.java)
+        assertEquals(EventName.EventErrored, e.name)
+        assertTrue(e.error!!.contains("Cannot deserialize"))
     }
 
     @Test
     fun `event is published to retry queue on CordaError`() {
         //when
         `when`(cordaService.consentBranchExists(any())).thenThrow(CordaRuntimeException("boom!"))
-        var event = Serialization.objectMapper().writeValueAsString(newConsentRequestStateAsEvent())
+        val event = Serialization.objectMapper().writeValueAsString(newConsentRequestStateAsEvent())
         connection.publish(NATS_CONSENT_REQUEST_SUBJECT, event.toByteArray())
 
         Thread.sleep(2000)
@@ -188,13 +193,17 @@ class NatsToCordaPipelineIntegrationTest {
     fun `event is published to error queue on unknown exception`() {
         //when
         `when`(cordaService.consentBranchExists(any())).thenThrow(IllegalArgumentException("boom!"))
-        var event = Serialization.objectMapper().writeValueAsString(newConsentRequestStateAsEvent())
+        val event = Serialization.objectMapper().writeValueAsString(newConsentRequestStateAsEvent())
         connection.publish(NATS_CONSENT_REQUEST_SUBJECT, event.toByteArray())
 
         Thread.sleep(2000)
 
         // then an entry must be available in the retry queue
         assertEquals(1, erroredEvents.size)
+
+        val e: Event = Serialization.objectMapper().readValue(erroredEvents.first().byteInputStream(), Event::class.java)
+        assertEquals(EventName.EventErrored, e.name)
+        assertEquals("boom!", e.error)
     }
 
     private fun newConsentRequestStateAsEvent() : Event {

